packages/react-relay/__tests__/LiveResolvers-test.js

```diff
@@ -1006,3 +1006,14 @@ test('with client-only field and args', () => {
   });
   expect(renderer.toJSON()).toEqual('Counter is 2');
 });
+
+test('Can read a live client edge without a fragment', () => {
+  // ... this panics in the compiler
+  const FooQuery = graphql`
+    query LiveResolversTest13Query {
+      live_constant_client_edge @waterfall {
+        name
+      }
```

Comment:
oh... I'm not sure how can we generate a query for this selection if we don't have a fragment for this resolver

Reply:
This is much better. I have this D36971289 as patch in the compiler, but your version is much better. Thanks for fixing this!